src/test/java/com/desiato/puresynth/controllers/LoginControllerTest.java

```diff
@@ -0,0 +1,92 @@
+package com.desiato.puresynth.controllers;
+
+import com.desiato.puresynth.models.LoginDTO;
+import com.desiato.puresynth.models.User;
+import com.desiato.puresynth.repositories.UserRepository;
+import com.desiato.puresynth.services.CustomAuthenticationService;
+import com.desiato.puresynth.services.UserService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.UUID;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
```

Comment:
Right now, as you have 2 Spring test classe you should create a single abstract `BaseTest` which has those annotations and configuration. It's expensive to start a spring context, so your test will run faster with just one context (a test with `@SpringBootTest`).